pkg/cassandra/management.go

```diff
@@ -146,3 +155,41 @@ func (r *defaultManagementApiFacade) getCassandraContainerStatus(pod corev1.Pod)
 	}
 	return nil
 }
+
+func (r *defaultManagementApiFacade) ListKeyspaces(
+	keyspaceName string,
+) ([]string, error) {
+	if pods, err := r.fetchDatacenterPods(); err != nil {
+		r.logger.Error(err, "Failed to fetch datacenter pods")
+		return []string{}, err
+	} else {
+		for _, pod := range pods {
+			if keyspaces, err := r.nodeMgmtClient.GetKeyspace(&pod, keyspaceName); err != nil {
+				r.logger.Error(err, fmt.Sprintf("Failed to CALL list keyspaces %s on pod %v", keyspaceName, pod.Name))
+			} else {
+				return keyspaces, nil
+			}
+		}
+		return []string{}, fmt.Errorf("CALL list keyspaces %s failed on all datacenter %v pods", keyspaceName, r.dc.Name)
+	}
+}
+
+func (r *defaultManagementApiFacade) AlterKeyspace(
+	keyspaceName string,
+	replicationSettings map[string]int,
+) error {
+	if pods, err := r.fetchDatacenterPods(); err != nil {
+		r.logger.Error(err, "Failed to fetch datacenter pods")
+		return err
+	} else {
+		for _, pod := range pods {
+			if err := r.nodeMgmtClient.AlterKeyspace(&pod, keyspaceName, r.createReplicationConfig(replicationSettings)); err != nil {
```

Comment:
Just curious, what happens if the keyspace exists but was created with SimpleStrategy? Does the alter fail?

Reply:
no it will succeed. You can alter SimpleStrategy to NTS, but it can generate some inconsistencies without a proper repair.